src/protocols/secure_channel/tests/TestPASESession.cpp

```diff
@@ -41,22 +41,53 @@ using namespace chip::Protocols;
 
 using TestContext = chip::Test::MessagingContext;
 
+static void test_os_sleep_ms(uint64_t millisecs)
+{
+    struct timespec sleep_time;
+    uint64_t s = millisecs / 1000;
+
+    millisecs -= s * 1000;
+    sleep_time.tv_sec  = static_cast<time_t>(s);
+    sleep_time.tv_nsec = static_cast<long>(millisecs * 1000000);
+
+    nanosleep(&sleep_time, nullptr);
+}
+
 class LoopbackTransport : public Transport::Base
 {
 public:
     CHIP_ERROR SendMessage(const PeerAddress & address, System::PacketBufferHandle && msgBuf) override
     {
         ReturnErrorOnFailure(mMessageSendError);
         mSentMessageCount++;
-        HandleMessageReceived(address, std::move(msgBuf));
+
+        if (mNumMessagesToDrop == 0)
+        {
+            System::PacketBufferHandle receivedMessage = msgBuf.CloneData();
+            HandleMessageReceived(address, std::move(receivedMessage));
```

Comment:
Is there a reason we clone instead of just moving out of `msgBuf`?  Worth documenting.

Reply:
I'll document it. The main reason is that the original buffer is used for retransmission, and if the same buffer is used for receive, it manipulates it for extracting headers etc. So it cannot be used for retransmission afterwards.